File: backend/app/schemas/member.py
from __future__ import annotations

from pydantic import BaseModel, ConfigDict


class MemberCreate(BaseModel):
    first_name: str
    last_name: str
    email: str
    linkedin_url: str
    phone: str | None = None
    record_source: str | None = None  # 'Framer' | 'Luma' | 'Substack' | 'Manual'


class MemberUpdate(BaseModel):
    first_name: str | None = None
    last_name: str | None = None
    email: str | None = None
    linkedin_url: str | None = None
    phone: str | None = None
    record_source: str | None = None


class MemberResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    first_name: str
    last_name: str
    email: str
    linkedin_url: str
    phone: str | None = None
    record_source: str | None = None

File: backend/src/schemas/member.ts
import { z } from "zod";

export const MemberCreateSchema = z.object({
  first_name: z.string(),
  last_name: z.string(),
  email: z.string(),
  linkedin_url: z.string(),
  phone: z.string().nullable().optional(),
  record_source: z.string().nullable().optional(),
});
export type MemberCreate = z.infer<typeof MemberCreateSchema>;

export const MemberUpdateSchema = MemberCreateSchema.partial();
export type MemberUpdate = z.infer<typeof MemberUpdateSchema>;

export const MemberResponseSchema = MemberCreateSchema.extend({
  id: z.string(),
});
export type MemberResponse = z.infer<typeof MemberResponseSchema>;
